### PR TITLE
context `make_or_load_nwbfile` to raise an error when nwbfile needs to be created but metadata is None

### PR DESCRIPTION
**Pull request by h-mayorquin**
Related to the issue #936

### CODE REVIEW

File: tests/test_minimal/test_tools/test_context_tools.py [hunk 1]
```diff
@@ -220,3 +221,11 @@ def test_make_or_load_nwbfile_on_corrupt_file(tmpdir: Path) -> None:
...
         time_series = mock_TimeSeries()
         nwbfile.add_acquisition(time_series)
+
+
+def test_raise_error_when_metadata_is_empty_and_creation_is_needed(tmpdir):
+    nwbfile_path = tmpdir / "test_make_or_load_nwbfile_empty_metadata.nwb"
+
+    with pytest.raises(ValueError):
+        with make_or_load_nwbfile(nwbfile_path=nwbfile_path, metadata=None, overwrite=True) as nwbfile:
+            nwbfile
```
Comment (CodyCBakerPhD) [inline tests/test_minimal/test_tools/test_context_tools.py:231]:
```suggestion
            pass
```

I think you can just do this?

Comment (h-mayorquin) [reply tests/test_minimal/test_tools/test_context_tools.py:231]: Yes both work, do you prefer the latter?

**Comment by h-mayorquin** [comment]
Dandi seems to be asking us to upgrade?

File: tests/test_minimal/test_tools/test_context_tools.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply tests/test_minimal/test_tools/test_context_tools.py:231]: I prefer the latter unless you want to do something with the object (print, assert something, etc)

**Comment by CodyCBakerPhD** [comment]
> Dandi seems to be asking us to upgrade?

They did a release today and it might have put their side out of sync. We'll keep an eye on it but it shouldn't have anything to do with our side of things

File: tests/test_minimal/test_tools/test_context_tools.py (hunk 1 above)
Comment (h-mayorquin) [reply tests/test_minimal/test_tools/test_context_tools.py:231]: OK, let's change it. I am adding a changelog and changing to pass.

Comment (h-mayorquin) [reply tests/test_minimal/test_tools/test_context_tools.py:231]: Done.

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**